torchgeo/datasets/spacenet.py

```diff
@@ -191,49 +199,287 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        # Check if collections exist
+        missing_collections = []
+        for collection in self.collections:
+            stacpath = os.path.join(self.root, collection, "collection.json")
+
+            if not os.path.exists(stacpath):
+                missing_collections.append(collection)
+
+        if not missing_collections:
+            return []
+
+        to_be_downloaded = []
+        for collection in missing_collections:
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if os.path.exists(archive_path):
+                print(f"Found {collection} archive")
+                if (
+                    self.checksum
+                    and check_integrity(
+                        archive_path, self.collection_md5_dict[collection]
+                    )
+                    or not self.checksum
+                ):
+                    print("Extracting...")
+                    extract_archive(archive_path)
+                else:
+                    print(f"Collection {collection} is corrupted")
+                    to_be_downloaded.append(collection)
+            else:
+                print(f"{collection} not found")
+                to_be_downloaded.append(collection)
+
+        return to_be_downloaded
+
+    def _download(self, collections: List[str], api_key: Optional[str] = None) -> None:
         """Download the dataset and extract it.
 
         Args:
+            collections: Collections to be downloaded
             api_key: a RadiantEarth MLHub API key to use for downloading the dataset
 
         Raises:
             RuntimeError: if download doesn't work correctly or checksums don't match
         """
-        if self._check_integrity():
-            print("Files already downloaded")
-            return
-
-        download_radiant_mlhub_dataset(self.dataset_id, self.root, api_key)
-        archive_path = os.path.join(self.root, self.foldername + ".tar.gz")
-        if (
-            self.checksum
-            and check_integrity(archive_path, self.md5)
-            or not self.checksum
-        ):
+        for collection in collections:
+            download_radiant_mlhub_collection(collection, self.root, api_key)
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if (
+                not self.checksum
+                or not check_integrity(
+                    archive_path, self.collection_md5_dict[collection]
+                )
+            ) and self.checksum:
+                raise RuntimeError(f"Collection {collection} corrupted")
+
+            print("Extracting...")
             extract_archive(archive_path)
-        else:
-            raise RuntimeError("Dataset corrupted")
+
+
+class SpaceNet1(SpaceNet):
+    """SpaceNet 1: Building Detection v1 Dataset.
+
+    `SpaceNet 1 <https://spacenet.ai/spacenet-buildings-dataset-v1/>`_
+    is a dataset of building footprints over the city of Rio de Janeiro.
+
+    * *Dataset features*:
```

Comment:
I wouldn't make this a nested bulleted list, I think Dataset features and Dataset format can be text-only without a list.

Reply:
Do you mean to combine the list items as a paragraph of sorts?

Reply:
I mean do the same thing we're doing for other datasets, where it's only a list, not a nested list: https://torchgeo.readthedocs.io/en/latest/api/datasets.html#advance-audio-visual-aerial-scene-recognition-dataset